rdata/unparser/_unparser.py

```diff
@@ -0,0 +1,170 @@
+"""Abstract base class for unparsers."""
+
+from __future__ import annotations
+
+import abc
+from typing import TYPE_CHECKING, Any
+
+import numpy as np
+
+from rdata.parser import (
+    RData,
+    RExtraInfo,
+    RObject,
+    RObjectInfo,
+    RObjectType,
+    RVersions,
+)
+
+if TYPE_CHECKING:
+    import numpy.typing as npt
+
+
+def pack_r_object_info(info: RObjectInfo) -> np.int32:
+    """Pack RObjectInfo to an integer."""
+    if info.type == RObjectType.NILVALUE:
+        bits = f"{0:24b}"
+    elif info.type == RObjectType.REF:
+        bits = f"{info.reference:24b}"
+    else:
+        bits = (f"{0:4b}"
+                f"{info.gp:16b}"
+                f"{0:1b}"
+                f"{info.tag:1b}"
+                f"{info.attributes:1b}"
+                f"{info.object:1b}"
+                )
```

Comment:
Small formatting nitpick (so that everything is aligned and parentheses match):
```suggestion
        bits = (
            f"{0:4b}"
            f"{info.gp:16b}"
            f"{0:1b}"
            f"{info.tag:1b}"
            f"{info.attributes:1b}"
            f"{info.object:1b}"
        )
```

Reply:
Fixed.